addons/portal/static/src/js/skip_to_content.js

```diff
@@ -0,0 +1,58 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+
+publicWidget.registry.SkipToContent = publicWidget.Widget.extend({
+    selector: ".o_skip_to_content",
+
+    /**
+     * @override
+     */
+    start() {
+        this._observeBodyClass();
+        this._toggleSkipToContent();
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this.observer?.disconnect();
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _toggleSkipToContent() {
+        // The button should not be visible when there is no header present, as
+        // there would be nothing to skip. It should also be hidden when a modal is
+        // open.
+        const isModalPresent = document.querySelector("body")?.classList.contains("modal-open");
+        const hasNoHeader = document.querySelector("header")?.classList.contains("d-none");
```

Comment:
I don't get this either. 1) The xpath adds the skip link only before `<header>`, so we know there is one, 2) the skip link will be positioned in the top left of the page anyway: so when the header in not shown (likely after scrolling), the skip link won't either.
Why do you think this is needed?